quantmsio/core/feature.py

```
@@ -1,535 +1,190 @@
+        ]
+
+        def merge_psm(rows, index):
+            key = (rows["reference_file_name"], rows["peptidoform"], rows["precursor_charge"])
+            if key in map_dict:
+                return map_dict[key][index]
+            else:
+                return None
+
+        for i, feature in enumerate(map_features):
+            msstats.loc[:, feature] = msstats[["reference_file_name", "peptidoform", "precursor_charge"]].apply(
+                lambda rows: merge_psm(rows, i),
```

Ruff

Function definition does not bind loop variable `i`

(B023)